File: SocialNetwork.WebApp/src/types/comment.ts
import { MediaType } from "../enums/media";
import { CommentMentionPagination } from "../utils/pagination";
import { BaseResponse } from "./response";
import { UserResource } from "./user";

export type CommentResource = {
    id: string; 
    content: string;
    postId: string;
    parentCommentId?: string | null;
    replyToUserId?: string | null;
    replyToUserName?: string | null; 
    mediaUrl?: string; 
    mediaType?: MediaType; 
    user: UserResource;
    createdAt: Date; 
    sentAt: Date; 
    isHaveChildren: boolean;
    replies: CommentResource[];
    status: string;
    level: number;
    pagination?: CommentMentionPagination
};

export interface CommentMentionPaginationResource extends BaseResponse {
    data: CommentResource[];
    pagination: CommentMentionPagination
}

File: SocialNetwork.Domain/Entity/Comment.cs
using SocialNetwork.Domain.Abstractions;


namespace SocialNetwork.Domain.Entity
{
    public class Comment : BaseAuditableEntity
    {
        public string? Content { get; set; }
        public string? MediaType { get; set; }
        public string? MediaUrl { get; set; }
        public string? ReplyToUserId { get; set; }
        public string? ReplyToUserName { get; set; }
        public Guid? ParentCommentId { get; set; }
        public Comment ParentComment { get; set; }
        public Guid PostId { get; set; }
        public Post Post { get; set; }
        public string UserId { get; set; } // Who create comment
        public User User { get; set; } // Who create comment
        public ICollection<Comment> Replies { get; set; }
    }
}
